addons/account/models/account_payment.py

```diff
@@ -489,187 +674,35 @@ def post(self):
                             sequence_code = 'account.payment.supplier.refund'
                         if rec.payment_type == 'outbound':
                             sequence_code = 'account.payment.supplier.invoice'
-                rec.name = self.env['ir.sequence'].with_context(ir_sequence_date=rec.payment_date).next_by_code(sequence_code)
+                rec.name = self.env['ir.sequence'].next_by_code(sequence_code, sequence_date=rec.payment_date)
                 if not rec.name and rec.payment_type != 'transfer':
                     raise UserError(_("You have to define a sequence for %s in your company.") % (sequence_code,))
 
-            # Create the journal entry
-            amount = rec.amount * (rec.payment_type in ('outbound', 'transfer') and 1 or -1)
-            move = rec._create_payment_entry(amount)
+            moves = AccountMove.create(rec._prepare_payment_moves())
+            moves.filtered(lambda move: not move.journal_id.post_at_bank_rec).post()
+
+            # Update the state / move before performing any reconciliation.
+            rec.write({'state': 'posted', 'move_name': moves[0].name})
 
-            # In case of a transfer, the first journal entry created debited the source liquidity account and credited
-            # the transfer account. Now we debit the transfer account and credit the destination liquidity account.
-            if rec.payment_type == 'transfer':
-                transfer_credit_aml = move.line_ids.filtered(lambda r: r.account_id == rec.company_id.transfer_account_id)
-                transfer_debit_aml = rec._create_transfer_entry(amount)
-                (transfer_credit_aml + transfer_debit_aml).reconcile()
+            if rec.payment_type in ('inbound', 'outbound'):
+                # ==== 'inbound' / 'outbound' ====
+                if rec.invoice_ids:
+                    (moves[0] + rec.invoice_ids).mapped('line_ids')\
```

Comment:
`.mapped()` for relational fields is deprecated

Reply:
not deprecated, just recommended :)

I'm sure there are dozens places when you could drop the same comment.